File: frontend/src/types/character.ts
export interface Character {
  id: number;
  novelId: number;
  name: string;
  role: string | null;
  age: number | null;
  personality: string | null;
  speechStyle: string | null;
  description: string | null;
  isFavorite: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CharacterCreateRequest {
  name: string;
  role?: string;
  age?: number;
  personality?: string;
  speechStyle?: string;
  description?: string;
}

export interface CharacterUpdateRequest {
  name: string;
  role?: string;
  age?: number;
  personality?: string;
  speechStyle?: string;
  description?: string;
}

File: src/main/java/org/example/domain/character/entity/Character.java
package org.example.domain.character.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.example.domain.novel.entity.Novel;
import org.example.global.common.BaseEntity;

@Entity
@Table(name = "characters")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Character extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "novel_id", nullable = false)
    private Novel novel;

    @Column(nullable = false)
    private String name;

    @Column
    private String role;

    @Column
    private Integer age;

    @Column(columnDefinition = "TEXT")
    private String personality;

    @Column(columnDefinition = "TEXT")
    private String speechStyle;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, columnDefinition = "TINYINT(1) DEFAULT 0")
    private Boolean isFavorite = false;

    @Builder
    private Character(Novel novel, String name, String role, Integer age,
                      String personality, String speechStyle, String description) {
        this.novel = novel;
        this.name = name;
        this.role = role;
        this.age = age;
        this.personality = personality;
        this.speechStyle = speechStyle;
        this.description = description;
        this.isFavorite = false;
    }

    public void update(String name, String role, Integer age,
                       String personality, String speechStyle, String description) {
        this.name = name;
        this.role = role;
        this.age = age;
        this.personality = personality;
        this.speechStyle = speechStyle;
        this.description = description;
    }

    public void updateFavorite(boolean isFavorite) {
        this.isFavorite = isFavorite;
    }
}
